modules/tests/shared/src/test/scala/io/circe/JavaTimeCodecSuite.scala

```diff
@@ -25,6 +26,28 @@ class JavaTimeCodecSuite extends CirceMunitSuite {
   private[this] val minInstant: Instant = Instant.EPOCH
   private[this] val maxInstant: Instant = Instant.parse("3000-01-01T00:00:00.00Z")
 
+  // Backported from ScalaCheck
+  // https://github.com/typelevel/scalacheck/blob/v1.16.0/core/shared/src/main/scala/org/scalacheck/time/JavaTimeChoose.scala#L225
+  private[this] implicit final lazy val chooseZoneOffset: Choose[ZoneOffset] =
+    new Choose[ZoneOffset] {
+      override def choose(min: ZoneOffset, max: ZoneOffset): Gen[ZoneOffset] =
+        min.compareTo(max) match {
+          case 0                    => Gen.const(min)
+          case result if result > 0 => Gen.fail
+          case _                    =>
+            // Looks flipped, but it is not.
+            Gen.choose(max.getTotalSeconds, min.getTotalSeconds).map(value => ZoneOffset.ofTotalSeconds(value))
+        }
+    }
```

Comment:
I'm fine with bumping scalacheck to 1.16.0 if we need to so we don't have to do this backport.

Reply:
We need ScalaCheck 1.17.0 actually.

Reply:
Ok, we can keep this backported until 1.17.x then.